docs/configuration.md

```
@@ -1,8 +1,318 @@
+> **Note**: The application automatically detects and converts legacy snake_case configurations for backward compatibility.
+
+## Configuration Architecture
+
+### Philosophy: Single File, Logical Sections
+**Reject complexity. Embrace simplicity.**
+
+Instead of multiple configuration files that create cognitive overhead, use a **single, well-structured configuration file** with logical sections.
+
+### Unified Configuration Structure
+
+```
```

markdownlint-cli2 (0.17.2)

Fenced code blocks should have a language specified

(MD040, fenced-code-language)